### PR TITLE
Update where and import-sorting guides

### PR DESCRIPTION
**Pull request by pbrisbin**
See distinct commits.

### CODE REVIEW

**Review by jdreaver — APPROVED** [review]
Eh I think 1-spaced `where` is weird but I see the logic and I can get on board.

**Comment by cbeav** [comment]
Agreed I thought this was weird too, but we're actually already using it in a surprising number of places in the codebase. I'm down for whatever as long as we're consistent, and I do like the vertical alignment above and below the `where` line.

**Comment by pbrisbin** [comment]
Same here. I had a tough time acclimating to this, but I'm OK with it now. In my own 4-space world, this same style works out to 2-space/4-space instead of 1-space/2-space, so it's a little less weird.

Just to clarify, I'm not introducing this style here; it was already our preference. My intent here is to stop allowing inconsistencies and have the guide specify only one style.

**Review by cbeav — APPROVED** [review]

**Comment by eborden** [comment]
Yeah, I also thought 1 space was odd at first, but I liked the alignment and context free love it brings.

Normal expression:
```
foo =
  bar
 where
  bar = "baz"
```

`do` notation:
```
foo = do
  bar
  bats
 where
  bar = putStrLn "baz"
```

**Review by eborden — APPROVED** [review]

**Review by 5outh — APPROVED** [review]
I'm down with these changes. Good example coverage.

File: haskell-style.md [hunk 1]
```diff
@@ -246,41 +246,36 @@ someBinding'' mx =
...
-   where -- PARSE ERROR - parser thinks this where is a part of the last expression
-    f x y = x * x + y * y
-
--- Instead, you have to do either this
-someBinding''' mx my = do
-  x <- mx
-  y <- my
-  f x y
+-- Good - use the same indentation for non-multi-line expressions to avoid
+-- having to change anythign if/when they grow more lines
```
Comment (cdparks) [inline haskell-style.md]: `anythign` -> `anything`

**Review by cdparks — APPROVED** [review]

**Pull request merged**